openrouteservice/src/main/java/org/heigit/ors/routing/RouteResultBuilder.java

```diff
@@ -321,7 +322,7 @@ else if (sign == 1)
 	}
 
 	private int getEndWayPointIndex(int startIndex, InstructionType instrType, Instruction instr) {
-		if (instrType == InstructionType.FINISH)
+		if (instrType == InstructionType.FINISH || (instrType == InstructionType.DEPART && instr.getDistance() == 0.0 && instr.getPoints().size() == 1))
```

Comment:
Can we make this more readable? I think it is not really clear, what this condition means and why it covers the issue to be fixed. For example, it is unintuitive that the conjunction contains `InstructionType.DEPART` although the unit test has identical coordinates in the middle.

Reply:
Yes, but the way GH returns points for many cases is unintuitive and all conditions are required to not let some other cases fail. All I could think of is add some inline docs...

Reply:
>  it is unintuitive that the conjunction contains `InstructionType.DEPART` although the unit test has identical coordinates in the middle.

and that's because ORS requests each segment of the route separately to GH, so two consecutive identical coordinates are handled as one query regardless where they occur in a list of coordinates.